botocore/docs/bcdoc/style.py

```diff
@@ -249,7 +249,13 @@ def end_a(self):
                 self.doc.hrefs[self.a_href] = self.a_href
                 self.doc.write('`__')
             self.a_href = None
-        self.doc.write(' ')
+
+        special_characters = ['.', ',', '?', '!', ':', ';']
```

Comment:
Rather than declaring this list every time it's called, let's move it to a module constant so it's only built once. We can also use a tuple instead of a list it make it immutable.